src/heartratetask/HeartRateTask.h

```diff
@@ -16,10 +19,33 @@ namespace Pinetime {
   namespace Applications {
     class HeartRateTask {
     public:
-      enum class Messages : uint8_t { GoToSleep, WakeUp, StartMeasurement, StopMeasurement };
-      enum class States { Idle, Running };
+      enum class Messages : uint8_t {
+        // Screen gets turned off
+        GoToSleep,
+        // Screen gets turned on
+        WakeUp,
+        // Start button pressed
+        StartMeasurement,
+        // Stop button pressed
+        StopMeasurement
+      };
+
+      enum class States {
+        // Screen turned off, heartrate not measured
+        Idle,
+        // Screen turned on, heartrate app open, heartrate not measured
+        Running,
+        // Screen turned on, heartrate app open, heartrate actively measured
+        Measuring,
+        // Screen turned off, heartrate task is waiting until the next measurement should be started
+        BackgroundWaiting,
+        // Screen turned off, heartrate actively measured
+        BackgroundMeasuring
+      };
```

Comment:
I think `Running` is confusing, because it's the state where the measurement is not running. 

What do you think about those?
```
BackgroundIdle
ForegroundIdle
ForegroundMeasuring
BackgroundWaiting
BackgroundIdle
```

I don't know if I get all the states correctly, but `ForegroundWaiting` does not exist? Otherwise we could split the states into two enums (at least logically).

Reply:
`ForegroundRunning` (or just `Running`) means that you have the App open and you watch for the number on the screen.. in that moment you want to do continuously measure the heartrate and therefore there is no reason to have a `ForegroundWaiting`

Reply:
Makes sense. But according to the comments, `Running` is the state without active measuring.

Reply:
yeah sorry, you are right, it is not running in that way... I guess it is mainly because it was used before for that and the HeartRateController is using this state as well. I understand the confusion and maybe it would be good to change it, but it also changes the whole thing further :thinking: .. I am torn.

Reply:
Let's not keep technical debt because of convenience. This is our chance to give it a more speaking identifier.

Reply:
I actually agree that technical debt should be cleaned up right now when we have the chance. Otherwise it will probably never be cleaned.
I chose some better names (I think). But if you have other suggestions, it's easily changed